maxim/logger/smolagents/client.py

```
@@ -0,0 +1,539 @@
+        for attr in ['temperature', 'max_tokens', 'top_p', 'frequency_penalty', 'presence_penalty']:
+            if hasattr(model, attr):
+                model_parameters[attr] = getattr(model, attr)
+        
+        # Check for kwargs or additional parameters
+        if hasattr(model, 'kwargs') and isinstance(model.kwargs, dict):
+            model_parameters.update(model.kwargs)
+    
+    return messages, model_parameters
+
+
+def extract_model_info(agent):
```

Ruff (0.12.2)

`extract_model_info` is too complex (12 > 10)

(C901)